### PR TITLE
HDDS-3408. Rename ChunkLayOutVersion to ContainerLayoutVersion.

### PR DESCRIPTION
**Pull request by JyotinderSingh**
## What changes were proposed in this pull request?

Because this layOutversion defines/handles any changes related to chunks/blocks/container data. So, this layOutVersion on whole handles the entire Container. This jira proposes to rename ChunkLayOutVersion -> ContainerLayoutVersion. In this way, it will provide a clear explanation of this layOutVersion field.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-3408

## How was this patch tested?

Related integration and unit tests.

### CODE REVIEW

**Comment by JyotinderSingh** [comment]
Hi @bharatviswa504 I found this unassigned Jira created by you and decided to take it up. Is this patch still required?

File: hadoop-hdds/common/src/main/resources/ozone-default.xml [hunk 1]
```diff
@@ -722,11 +722,11 @@
     </description>
   </property>
   <property>
-    <name>ozone.scm.chunk.layout</name>
+    <name>ozone.scm.container.layout</name>
```
Comment (adoroszlai) [inline hadoop-hdds/common/src/main/resources/ozone-default.xml:725]: Changing config property names needs to be backward compatible by linking the old and new ones:

https://github.com/apache/ozone/blob/d09cdd4a585b66e95f32c8a054ee493101ee4de0/hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/conf/OzoneConfiguration.java#L299-L309

**Comment by bharatviswa504** [comment]
> Hi @bharatviswa504 I found this unassigned Jira created by you and decided to take it up. Is this patch still required?

@JyotinderSingh It is good to have, but as it involves config change we need to handle it in a backward compatible manner.

File: hadoop-hdds/common/src/main/resources/ozone-default.xml (hunk 1 above)
Comment (JyotinderSingh) [reply hadoop-hdds/common/src/main/resources/ozone-default.xml:725]: Thank you for letting me know this, I have addressed this issue in my latest commit.

**Comment by JyotinderSingh** [comment]
Thank you for the reviews @adoroszlai @bharatviswa504. I have linked the old config key to the new one.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/ContainerData.java [hunk 2]
```diff
@@ -225,11 +226,11 @@ public long getMaxSize() {
   }
 
   /**
-   * Returns the layOutVersion of the actual container data format.
-   * @return layOutVersion
+o  * Returns the layoutVersion of the actual container data format.
```
Comment (adoroszlai) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/ContainerData.java:229]: Stray character:

```suggestion
   * Returns the layoutVersion of the actual container data format.
```

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/ContainerData.java [hunk 3]
```diff
@@ -129,19 +129,20 @@
...
-      String originNodeId) {
+                          ContainerLayoutVersion layoutVersion, long size,
+                          String originPipelineId,
+                          String originNodeId) {
     Preconditions.checkNotNull(type);
 
     this.containerType = type;
     this.containerID = containerId;
-    this.layOutVersion = layOutVersion.getVersion();
+    this.layOutVersion = layoutVersion.getVersion();
```
Comment (adoroszlai) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/ContainerData.java:145]: Nit: Please also rename `this.layOutVersion` to `this.layoutVersion`.

Also in `DatanodeVersionFile`.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/ContainerDataYaml.java [hunk 4]
```diff
@@ -252,8 +252,9 @@ public Object construct(Node node) {
 
         //Needed this, as TAG.INT type is by default converted to Long.
         long layOutVersion = (long) nodes.get(OzoneConsts.LAYOUTVERSION);
```
Comment (adoroszlai) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/ContainerDataYaml.java:254]: Nit: for consistency.

```suggestion
        long layoutVersion = (long) nodes.get(OzoneConsts.LAYOUTVERSION);
```

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/ContainerDataYaml.java [hunk 5]
```diff
@@ -252,8 +252,9 @@ public Object construct(Node node) {
...
-        ChunkLayOutVersion layoutVersion =
-            ChunkLayOutVersion.getChunkLayOutVersion((int) layOutVersion);
+        ContainerLayoutVersion layoutVersion =
+            ContainerLayoutVersion.getContainerLayoutVersion(
+                (int) layOutVersion);
```
Comment (adoroszlai) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/ContainerDataYaml.java:257]: Nit: for consistency.

```suggestion
                (int) layoutVersion);
```

**Review by adoroszlai — COMMENTED** [review]
Thanks @JyotinderSingh for the patch.  Noticed couple of typos and leftover `layOutVersion` occurrences, but otherwise looks good.  Nice job keeping the key written to the container data YAML file as `layOutVersion`.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/ContainerData.java (hunk 3 above)
Comment (JyotinderSingh) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/ContainerData.java:145]: The variable name `layOutVersion` itself is used to name the key inside the container data YAML file. Renaming this to `layoutVersion` will change the key persisted into the YAML to `"layoutVersion"` as well (potentially breaking backward compatibility).

Comment (adoroszlai) [reply hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/ContainerData.java:145]: You are right, I forgot about that.

**Review by adoroszlai — APPROVED** [review]
Thanks @JyotinderSingh for updating the patch.

I just realized that `DatanodeVersionFile` has nothing to do with the chunk/container layout version.  So it might have been better to avoid touching that class, to make the change more coherent.  Sorry for asking for that typo fix.  However, I think it's a safe change, as `DatanodeVersionFile` uses constants as keys instead of the member variables of the class.

**Review by errose28 — APPROVED** [review]
Thanks @JyotinderSingh lgtm.

**Pull request merged**